newIDE/app/src/UI/ClosableTabs.js

```diff
@@ -192,12 +189,15 @@ export function ClosableTab({
               >
                 <span
                   style={{
-                    ...styles.tabLabel,
+                    ...styles.tabLabelAndIcon,
+                    height: muiTheme.closableTabs.height,
                     color: textColor,
                     fontFamily: muiTheme.closableTabs.fontFamily,
                   }}
                 >
-                  {label}
+                  {icon}
+                  {icon && label ? <Spacer /> : null}
+                  <span style={styles.tabLabel}>{label}</span>
```

Comment:
this doesn't create a weird space with the span if there's no label?

Reply:
Good spot. It's ok because the span size is 0x0 if no label is provided, so visually it's ok. But in an effort to reduce useless DOM element in the document, I've added a check 👍